keras_cv/visualization/segmentation_mask.py

```diff
@@ -0,0 +1,146 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility function to help visualize batch of binary mask.
+"""
+from collections import defaultdict
+
+import tensorflow as tf
+
+from keras_cv.visualization.colors import colors
+
+RGB = 3
+
+
+def _blend(image1, image2, factor=0.4):
+    difference = image2 - image1
+    scaled = factor * difference
+    # Do addition in float.
+    blended = image1 + scaled
+
+    # We need to clip and then cast.
+    blended = tf.round(tf.clip_by_value(blended, 0.0, 255.0))
+    return blended
+
+
+def _map_color_on_mask(masks, color):
+
+    # check distinct mask color codes with color mapping.
+    distinct_mask_code = tf.unique(tf.reshape(masks, -1)).y
+    color_rgb = defaultdict(tuple)
+
+    if isinstance(color, str):
+        for idx in distinct_mask_code:
+            try:
+                color_rgb.update({int(idx): colors[color]})
+            except KeyError:
+                raise KeyError(
+                    f"{color} is not supported yet,"
+                    "please check supported colors at `keras_cv.visualization.colors`"
+                )
+
+        color_keys = list(distinct_mask_code.numpy())
+
+    else:
+        if any([code not in distinct_mask_code for code in color.keys()]):
+            raise ValueError(
+                f"Color mapping {color} does not map completely\
+                  with distint color codes present in masks: {distinct_mask_code}"
+            )
+
+        # map color code with RGB
+        for code, name in color.items():
+            try:
+                color_rgb.update({code: colors[name]})
+            except KeyError:
+                raise KeyError(
+                    f"{name} is not supported yet,"
+                    "please check supported colors at `keras_cv.visualization.colors`"
+                )
+        color_keys = list(color.keys())
+
+    keys_tensor = tf.constant(color_keys, dtype=tf.int32)
+    colored_masks = tf.TensorArray(
+        tf.int32, size=0, dynamic_size=True, clear_after_read=True
+    )
+
+    for c in range(RGB):
+        vals_tensor = tf.constant([color_rgb[color][c] for color in color_keys])
+        table = tf.lookup.StaticHashTable(
+            tf.lookup.KeyValueTensorInitializer(keys_tensor, vals_tensor),
+            default_value=0,
+        )
+        colored_masks.write(c, table[tf.cast(masks, tf.int32)]).mark_used()
+
+    return tf.transpose(colored_masks.stack(), list(range(1, masks.ndim + 1)) + [0])
+
+
+def draw_segmentation(image, mask, color={}, alpha=0.4):
+    """Draws segmentation masks on images with desired color
+    and transparency.
+    Colors supported are standard X11 colors.
+
+    Args:
+        image: An integer tensor with shape (N, img_height, img_height, 3)
```

Comment:
Can we support single images?  So also (img_height, img_width, 3)

Reply:
tf.image.draw_bounding_boxes_on image support only 4D images.
https://www.tensorflow.org/api_docs/python/tf/image/draw_bounding_boxes

We can support 3d images, but idk if we can restrict to 4D and let users add new batch dim on first dim.

Reply:
done